src/App.js

```diff
@@ -1,11 +1,80 @@
+import TodoList from './app/components/TodoList';
+import SignUp from './app/components/forms/SignUp';
+import LogIn from './app/components/forms/LogIn';
+import {
+  BrowserRouter as Router,
+  Switch,
+  Route,
+  Link,
+  Redirect
+} from "react-router-dom";
+
+import UserApi from './app/services/UserApi';
 var React = require('react');
 var ReactDOM = require('react-dom');
+
 export default class App extends React.Component {
+  constructor(props) {
+    super(props);
+    this.state = {
+      isTokenPresent: new UserApi().userTokenPresent()       
+    }
+  }
+  
   render() {
     return (
+      <Router>
       <div>
-        Hello Vetal
+        <nav>
+          {this.renderIfAuth()}
+        </nav>
+        <Switch>
+          <Route path="/sign_up">
+            {this.state.isTokenPresent ? <Redirect to="/"/> : <SignUp afterLoggedIn={()=>{this.syncToken()}}/> }            
+          </Route>
+          <Route path="/log_in">
+            {this.state.isTokenPresent ? <Redirect to="/"/>  : <LogIn afterLoggedIn={()=>{this.syncToken()}}/> }  
+          </Route>
+          <Route path="/">
+            {this.state.isTokenPresent && <TodoList />}
+          </Route>
+        </Switch>
       </div>
-    );
-  }
+    </Router>
+    )}
+
+    syncToken(){
+      let isToken = new UserApi().userTokenPresent()
+      this.setState({isTokenPresent: isToken})
+    }
+
+    renderIfAuth(){
+      if (!this.state.isTokenPresent){
+        return(
+          <ul>
+            <li>
+                <Link to="/sign_up">Sign up</Link>
```

Comment:
usually all text like "Sign up", "Log in" should be in variables like "TRANSLATION". Ideally in a separate file.